src/v0/sources/shopify/transform.js

```diff
@@ -191,25 +197,50 @@ const processEvent = async (inputEvent, metricMetadata) => {
   message = removeUndefinedAndNullValues(message);
   return message;
 };
-
-const isIdentifierEvent = (event) => event?.event === 'rudderIdentifier';
-
+const isIdentifierEvent = (event) => ['rudderIdentifier', 'rudderSessionIdentifier'].includes(event?.event);
 const processIdentifierEvent = async (event, metricMetadata) => {
   if (useRedisDatabase) {
-    const lineItemshash = getHashLineItems(event.cart);
-    const value = ['anonymousId', event.anonymousId, 'itemsHash', lineItemshash];
+    let value;
+    let field;
+    if (event.event === 'rudderIdentifier') {
+      field = 'anonymousId';
+      const lineItemshash = getHashLineItems(event.cart);
+      value = ['anonymousId', event.anonymousId, 'itemsHash', lineItemshash];
+      stats.increment('shopify_redis_calls', {
+        type: 'set',
+        field: 'itemsHash',
+        ...metricMetadata,
+      });
+      /* cart_token: {
+           anonymousId: 'anon_id1',
+           lineItemshash: '0943gh34pg'
+          }
+      */
+    } else {
+      field = 'sessionId';
+      value = ['sessionId', event.sessionId];
```

Comment:
Value should be array?

Reply:
Yes it is the expected redis input format for [hmset](https://redis.io/commands/hmset/) , [ioredis doc](https://github.com/redis/ioredis#:~:text=//%20hmset(%27key%27%2C%20%7B%20k1%3A%20%27v1%27%2C%20k2%3A%20%27v2%27%20%7D))

Reply:
Got it